src/tracers.jl

```diff
@@ -1,30 +1,19 @@
 abstract type AbstractTracer <: Real end
 
-sparse_vector(T, index::Integer) = T([index])
-
 #===================#
 # Set operations    #
 #===================#
 
-myempty(::Type{S}) where {S<:AbstractSet} = S()
-
-function clever_union(a::AbstractSet{I}, b::AbstractSet{I}) where {I}
-    if isempty(a)
-        return b
-    elseif isempty(b)
-        return a
-    else
-        return union(a, b)
-    end
-end
+@inline myempty(::Type{S}) where {S<:AbstractSet} = S()
```

Comment:
this took lots of time in the `britgas` benchmark for some configurations, so it might be worth revisiting constant empty sets at a later day

Reply:
See https://github.com/adrhill/SparseConnectivityTracer.jl/pull/119#discussion_r1642652914.

Reply:
Adding an `EmptyPattern` type was part of the idea of the original PR, but I guess that will be close enough.